app/controllers/api/v1/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Users
+      class OmniauthCallbacksController < Devise::OmniauthCallbacksController
+        include ErrorRenderable
+
+        rescue_from ArgumentError, with: :render_bad_request
+
+        def google_oauth2
+          @user = User.from_omniauth(auth)
```

Comment:
```suggestion
          @user = User.from_omniauth(request.env['omniauth.auth'])
```

It seems that we only need the "auth" call once. So maybe we do not need to assign it as an instance variable and to cache it? :)